webapp/tests/karma/ts/modules/analytics/analytics-target-aggregates-sidebar-filter-component.spec.ts

```diff
@@ -0,0 +1,144 @@
+        fixture.detectChanges();
+      });
+  }));
+
+  afterEach(() => {
+    store.resetSelectors();
+    sinon.restore();
+  });
+
+  it('should create component', fakeAsync(() => {
+    expect(component).to.exist;
+    expect(component.isOpen).to.be.false;
+    expect(component.error).to.be.undefined;
+  }));
+
+  it('should toggle sidebar filter', () => {
+    component.toggleSidebarFilter();
+    component.toggleSidebarFilter();
+    component.toggleSidebarFilter();
+
+    expect(globalActions.setSidebarFilter.calledThrice).to.be.true;
+    expect(globalActions.setSidebarFilter.args[0][0]).to.deep.equal({ isOpen: true });
+    expect(globalActions.setSidebarFilter.args[1][0]).to.deep.equal({ isOpen: false });
+    expect(globalActions.setSidebarFilter.args[2][0]).to.deep.equal({ isOpen: true });
+    expect(telemetryService.record.calledTwice).to.be.true;
+  });
+
+  it('should set userFacilities and selectedFacilityId when user has multiple facilities', fakeAsync(() => {
+    const userFacilities = [{ _id: 'facility_1' }, { _id: 'facility_2' }];
+    userSettingsService.getUserFacility.resolves(userFacilities);
+    userSettingsService.hasMultipleFacilities.resolves(true);
+
+    expect(component.userFacilities).to.deep.equal(userFacilities);
+    expect(component.selectedFacilityId).to.equal('facility_1');
+    expect(userSettingsService.getUserFacility.calledOnce).to.be.true;
+    expect(userSettingsService.hasMultipleFacilities.calledOnce).to.be.true;
+  }));
+
+  it('should set user facility name_key as facilityFilterLabel', fakeAsync(() => {
+    contactTypesService.getTypeId.returns('district_hospital');
+    component.userFacilities = [{ _id: 'facility_1', type: 'district_hospital' }];
+
+    expect(component.facilityFilterLabel).to.equal('District Hospital');
+    expect(settingsService.get.calledOnce).to.be.true;
+    expect(contactTypesService.getTypeId.calledOnce).to.be.true;
+  }));
+
+  it('should set error and default facilityFilterLabel when settings service fails', fakeAsync(() => {
+    const DEFAULT_FACILITY_LABEL = 'Facility';
+    settingsService.get.rejects({ some: 'err' });
+
+    component.ngOnInit();
+    flush();
+
+    expect(component.facilityFilterLabel).to.equal(DEFAULT_FACILITY_LABEL);
+    expect(component.error).to.be.true;
+  }));
+
+  it('should set error and default facilityFilterLabel when contact type is not found', fakeAsync(() => {
+    const DEFAULT_FACILITY_LABEL = 'Facility';
+    const settings = {
+      contact_types: [
+        { id: 'health_center', name_key: 'Health Center' }
+      ]
+    };
+    settingsService.get.resolves(settings);
+    contactTypesService.getTypeId.returns('district_hospital');
+    component.userFacilities = [{ _id: 'facility_1', type: 'district_hospital' }];
+
+    component.ngOnInit();
+    flush();
+
+    expect(component.facilityFilterLabel).to.equal(DEFAULT_FACILITY_LABEL);
+    expect(component.error).to.be.true;
+  }));
```

Comment:
Another test case, should not break when any of the other services throws error

Reply:
I don't fully understand this test case, could you please be more specific.

Reply:
a unit test to assert that it doesn't break when any of these throws errors 
settingsService.get
    contactTypesService.getTypeId